src/Core/Install.php

```diff
@@ -0,0 +1,326 @@
+		}
+		self::addCheck($checks, L10n::t('Generate encryption keys'), $res, true, $help);
+
+	}
+
+
+	private static function checkFunctions(&$checks) {
+		$ck_funcs = [];
+		self::addCheck($ck_funcs, L10n::t('libCurl PHP module'), true, true, "");
+		self::addCheck($ck_funcs, L10n::t('GD graphics PHP module'), true, true, "");
+		self::addCheck($ck_funcs, L10n::t('OpenSSL PHP module'), true, true, "");
+		self::addCheck($ck_funcs, L10n::t('PDO or MySQLi PHP module'), true, true, "");
+		self::addCheck($ck_funcs, L10n::t('mb_string PHP module'), true, true, "");
+		self::addCheck($ck_funcs, L10n::t('XML PHP module'), true, true, "");
+		self::addCheck($ck_funcs, L10n::t('iconv PHP module'), true, true, "");
+		self::addCheck($ck_funcs, L10n::t('POSIX PHP module'), true, true, "");
+
+		if (function_exists('apache_get_modules')) {
+			if (! in_array('mod_rewrite',apache_get_modules())) {
+				self::addCheck($ck_funcs, L10n::t('Apache mod_rewrite module'), false, true, L10n::t('Error: Apache webserver mod-rewrite module is required but not installed.'));
+			} else {
+				self::addCheck($ck_funcs, L10n::t('Apache mod_rewrite module'), true, true, "");
+			}
+		}
+
+		if (! function_exists('curl_init')) {
+			$ck_funcs[0]['status'] = false;
+			$ck_funcs[0]['help'] = L10n::t('Error: libCURL PHP module required but not installed.');
+		}
+		if (! function_exists('imagecreatefromjpeg')) {
+			$ck_funcs[1]['status'] = false;
+			$ck_funcs[1]['help'] = L10n::t('Error: GD graphics PHP module with JPEG support required but not installed.');
+		}
+		if (! function_exists('openssl_public_encrypt')) {
+			$ck_funcs[2]['status'] = false;
+			$ck_funcs[2]['help'] = L10n::t('Error: openssl PHP module required but not installed.');
+		}
+		if (! function_exists('mysqli_connect') && !class_exists('pdo')) {
+			$ck_funcs[3]['status'] = false;
+			$ck_funcs[3]['help'] = L10n::t('Error: PDO or MySQLi PHP module required but not installed.');
+		}
+		if (!function_exists('mysqli_connect') && class_exists('pdo') && !in_array('mysql', PDO::getAvailableDrivers())) {
+			$ck_funcs[3]['status'] = false;
+			$ck_funcs[3]['help'] = L10n::t('Error: The MySQL driver for PDO is not installed.');
+		}
+		if (! function_exists('mb_strlen')) {
+			$ck_funcs[4]['status'] = false;
+			$ck_funcs[4]['help'] = L10n::t('Error: mb_string PHP module required but not installed.');
+		}
+		if (! function_exists('iconv_strlen')) {
+			$ck_funcs[6]['status'] = false;
+			$ck_funcs[6]['help'] = L10n::t('Error: iconv PHP module required but not installed.');
+		}
+		if (! function_exists('posix_kill')) {
+			$ck_funcs[7]['status'] = false;
+			$ck_funcs[7]['help'] = L10n::t('Error: POSIX PHP module required but not installed.');
+		}
+
+		$checks = array_merge($checks, $ck_funcs);
+
+		// check for XML DOM Documents being able to be generated
+//		// todo XML Check currently not working
+//		try {
+//			$xml = new DOMDocument();

```

Comment:
I had to comment out the `DOMDocument` and `Exception` because it doesn't find the references. Do you have an idea why?

Reply:
Yes, when you declare a namespace at the top of the file, all the non-fully qualified class names will be matched against this namespace. Since `DOMDocument` and `Exception` are global classes, they need to be either written fully-qualified (`\DOMDocument` and `\Exception`) or added as `use` statements (`use DOMDocument;` and `use Exception;` respectively)